### PR TITLE
Remove incompatible flag from test

### PR DESCRIPTION
**Pull request by keith**
Fixes https://github.com/bazelbuild/rules_apple/issues/993

### CODE REVIEW

**Comment by segiddins** [comment]
Do we have any way to check the running bazel version in those shell tests?

**Comment by segiddins** [comment]
Added a commit that will hopefully smooth things over for bazel 3, @keith / @brentleyjones if y'all want to take another look

**Comment by segiddins** [comment]
@googlebot I consent

File: test/apple_shell_testutils.sh [hunk 1]
```diff
@@ -449,9 +449,14 @@ function do_action() {
       # Explicitly pass these flags to ensure the external testing infrastructure
       # matches the internal one.
       "--incompatible_merge_genfiles_directory"
-      "--incompatible_objc_compile_info_migration"
   )
 
+  local bazel_version="$(bazel --version)"
```
Comment (brentleyjones) [inline test/apple_shell_testutils.sh:454]: As part of the LTS support, didn't they say there was a way to know which ball version was being used?

As for this, if I run with a non-path bazel, this check will be wrong. Is there a way to get the path to the bazel that is running this?

Comment (segiddins) [reply test/apple_shell_testutils.sh:454]: further down, we already invoke bazel directly via `bazel`

**Review by segiddins — APPROVED** [review]

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**